### ISSUE TITLE
tacacs_client getting error with "-t chap" authenticate

### ISSUE THREAD

**Issue by vnguyen2015** [ansible/tacacs_plus#32]
I tested tacacs_client authenticate method with option '-t chap' and got error with "chap_ppp_id must be a string".
I seems like a problem in bin/tacacs_client line 109&110.
With following fix, it works for me:
args.authen_type == 'chap'
-->
args.authen_type == TAC_PLUS_AUTHEN_TYPES['chap']

**Comment by ryanpetrello** [comment]
@vnguyen2015 can you share how you're using `tacacs_client` (with passwords, etc... blanked out)?

**Comment by vnguyen2015** [comment]
@ryanpetrello I am setting up a tacacs server on a ubuntu 16.04 VM
The host run client as following commands
```
root@SPDM:/home/vinhnx/work# tacacs_client -u tac_user -H 10.10.1.181 -v -k testing123 -t ascii authenticate
password for tac_user: 
status: PASS
root@SPDM:/home/vinhnx/work# tacacs_client -u tac_user -H 10.10.1.181 -v -k testing123 -t pap authenticate
password for tac_user: 
status: PASS
root@SPDM:/home/vinhnx/work# tacacs_client -u tac_user -H 10.10.1.181 -v -k testing123 -t chap authenticate
password for tac_user: 
No handlers could be found for logger "__main__"
root@SPDM:/home/vinhnx/work# tacacs_client -u tac_user -H 10.10.1.181 -d -v -k testing123 -t chap authenticate
WARNING:__main__:TACACS+ --debug will log raw packet data INCLUDING PASSWORDS;proceed at your own risk!
password for tac_user: 
ERROR:__main__:error: chap_ppp_id must be a string
Traceback (most recent call last):
  File "/usr/local/bin/tacacs_client", line 160, in main
    sys.exit(authenticate(cli, args))
  File "/usr/local/bin/tacacs_client", line 116, in authenticate
    port=args.virtual_port)
  File "/usr/local/lib/python2.7/dist-packages/tacacs_plus/client.py", line 185, in authenticate
    raise ValueError('chap_ppp_id must be a string')
ValueError: chap_ppp_id must be a string
```

**Comment by vnguyen2015** [comment]
I have also found that it is missing *import socket* in bin/tacacs_client with recently changes

**Comment by ryanpetrello** [comment]
Thanks for reporting, @vnguyen2015.  This should all be fixed in the latest release on PyPI, 2.4.

https://pypi.python.org/pypi/tacacs_plus/2.4
`$ pip install tacacs_plus`
